Unable to include jenkins-build-monitor-plugin as dependency because of missing parent POM for jenkins-build-monitor

This issue is similar to #299 and [JENKINS-43171](https://issues.jenkins-ci.org/browse/JENKINS-43171).

We're using the [Job DSL plugin](https://github.com/jenkinsci/job-dsl-plugin) and [Gradle Jenkins Job DSL plugin](https://github.com/heremaps/gradle-jenkins-jobdsl-plugin) to automatically create Jenkins jobs. To generate the right XML, the Job DSL plugin uses the Jenkins plugin as dependency.

We are using Gradle to include dependencies, when we run Gradle it complains the parent POM is missing:
```
FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':jenkinsPlugins'.
> Could not resolve org.jenkins-ci.plugins:build-monitor-plugin:1.12+build.201708172343.
  Required by:
      project :
   > Could not resolve org.jenkins-ci.plugins:build-monitor-plugin:1.12+build.201708172343.
      > Could not parse POM https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/build-monitor-plugin/1.12+build.201708172343/build-monitor-plugin-1.12+build.201708172343.pom
         > Could not find org.jenkins-ci.plugins:build-monitor:1.12+build.201708172343.
           Searched in the following locations:
               https://repo1.maven.org/maven2/org/jenkins-ci/plugins/build-monitor/1.12+build.201708172343/build-monitor-1.12+build.201708172343.pom
               https://repo1.maven.org/maven2/org/jenkins-ci/plugins/build-monitor/1.12+build.201708172343/build-monitor-1.12+build.201708172343.jar
               https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/build-monitor/1.12+build.201708172343/build-monitor-1.12+build.201708172343.pom
               https://repo.jenkins-ci.org/releases/org/jenkins-ci/plugins/build-monitor/1.12+build.201708172343/build-monitor-1.12+build.201708172343.jar
               https://plugins.gradle.org/m2/org/jenkins-ci/plugins/build-monitor/1.12+build.201708172343/build-monitor-1.12+build.201708172343.pom
               https://plugins.gradle.org/m2/org/jenkins-ci/plugins/build-monitor/1.12+build.201708172343/build-monitor-1.12+build.201708172343.jar
```

This is the snippet of Gradle code where we include the plugin:
```
dependencies {
...
    jenkinsPlugins 'org.jenkins-ci.plugins:build-monitor-plugin:1.12+build.201708172343'
...
}
```

Indeed the parent POM is not in the Jenkins maven repository. Fix could be to either publish the parent POM or to refactor the plugin POM (this is suggested in [JENKINS-43171](https://issues.jenkins-ci.org/browse/JENKINS-43171)). Internally we have published the plugin and parent POMs in our private maven repository.

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -95,3 +95,8 @@ jobs:
             -Dfile=build-monitor-plugin/target/build-monitor-plugin.hpi \
             -DrepositoryId=repo.jenkins-ci.org \
             -Durl=https://repo.jenkins-ci.org/releases/
+          mvn --batch-mode deploy:deploy-file \
+            -DpomFile=pom.xml \
+            -Dfile=pom.xml \
+            -DrepositoryId=repo.jenkins-ci.org \
+            -Durl=https://repo.jenkins-ci.org/releases/

